tachyon/zk/plonk/examples/example_circuit_tests.cc

```diff
@@ -0,0 +1,527 @@
+                     math::bn254::G1AffinePoint>;
+
+using GWC = GWCExtension<math::bn254::BN254Curve, kMaxDegree,
+                         kMaxExtendedDegree, math::bn254::G1AffinePoint>;
+
+// Note(ashjeong): |Poly|, |Evals|, |Commitment|, and |Field| are currently the
+// same across all types
+using Halo2LS =
+    lookup::halo2::Scheme<typename SHPlonk::Poly, typename SHPlonk::Evals,
+                          typename SHPlonk::Commitment>;
+
+template <typename _Circuit, typename _PCS, typename _LS>
+struct TestingArguments {
+  using Circuit = _Circuit;
+  using PCS = _PCS;
+  using LS = _LS;
+};
+
+template <typename Circuit, typename PCS, typename LS>
+struct ExampleTestDataSelector {
+  using F = typename PCS::Field;
+
+  // clang-format off
+  using type =
+      std::conditional_t<
+        std::is_same_v<Circuit, SimpleCircuit<F, SimpleFloorPlanner>>,
+          SimpleTestData<SimpleCircuit<F, SimpleFloorPlanner>, PCS, LS>,
+          SimpleTestData<SimpleCircuit<F, V1FloorPlanner>, PCS, LS>>;
+  // clang-format on
+};
+template <typename TestingType>
+class ExampleCircuitTest
+    : public CircuitTest<typename TestingType::PCS, typename TestingType::LS> {
+ public:
+  static void SetUpTestSuite() { math::bn254::BN254Curve::Init(); }
+};
+
+using TestingTypes = testing::Types<
+    TestingArguments<SimpleCircuit<SHPlonk::Field, SimpleFloorPlanner>, SHPlonk,
+                     Halo2LS>,
+    TestingArguments<SimpleCircuit<SHPlonk::Field, V1FloorPlanner>, SHPlonk,
+                     Halo2LS>>;
+}  // namespace
+
+TYPED_TEST_SUITE(ExampleCircuitTest, TestingTypes);
+
+TYPED_TEST(ExampleCircuitTest, Configure) {
+  using TestingType = TypeParam;
+  using Circuit = typename TestingType::Circuit;
+  using PCS = typename TestingType::PCS;
+  using LS = typename TestingType::LS;
+  using F = typename PCS::Field;
+  using ExampleTestData =
+      typename ExampleTestDataSelector<Circuit, PCS, LS>::type;
+
+  ConstraintSystem<F> constraint_system;
+  auto config = Circuit::Configure(constraint_system);
+
+  ExampleTestData::TestConfig(config);
+
+  halo2::PinnedConstraintSystem<F> pinned_constraint_system(constraint_system);
+  EXPECT_EQ(ExampleTestData::kPinnedConstraintSystem,
+            base::ToRustDebugString(pinned_constraint_system));
+
+  EXPECT_TRUE(constraint_system.selector_map().empty());
+  EXPECT_TRUE(constraint_system.general_column_annotations().empty());
+}
+
+TYPED_TEST(ExampleCircuitTest, Synthesize) {
+  using TestingType = TypeParam;
+  using Circuit = typename TestingType::Circuit;
+  using PCS = typename TestingType::PCS;
+  using LS = typename TestingType::LS;
+  using F = typename PCS::Field;
+  using Domain = typename PCS::Domain;
+  using Commitment = typename PCS::Commitment;
+  using RationalEvals = typename PCS::RationalEvals;
+  using ExampleTestData =
+      typename ExampleTestDataSelector<Circuit, PCS, LS>::type;
+
+  CHECK(this->prover_->pcs().UnsafeSetup(ExampleTestData::kN, F(2)));
```

Comment:
Is it always 2?

Reply:
:thinking: I'm not sure in terms of the whole scope, but every previous example circuit test code had `F(2)`

Reply:
Okay let's think later on if we have cases that need degree greater than 2.